draw/src/font_atlas.rs

```diff
@@ -286,56 +287,95 @@ impl CxShapeCache {
         font_ids: &[usize],
         font_atlas: &CxFontAtlas,
         glyph_infos: &mut Vec<GlyphInfo>,
+        // Used to pass the current cluster index value after the font switch.
+        base_cluster: usize,
     ) -> Result<(), ()> {
-        let Some((&font_id, font_ids)) = font_ids.split_first() else {
+
+        // Get the preferred font to be used currently.
+        let Some((&font_id, remaining_font_ids)) = font_ids.split_first() else {
             return Err(());
         };
+
+        // Verify if the font is available, and if not, try the fallback font.
         let Some(font) = &font_atlas.fonts[font_id] else {
-            return self.shape_full_recursive(text, font_ids, font_atlas, glyph_infos);
+            return self.shape_full_recursive(
+                text,
+                remaining_font_ids,
+                font_atlas,
+                glyph_infos,
+                base_cluster
+            );
         };
 
+        // Create and configure the HarfBuzz buffer.
         let mut buffer = UnicodeBuffer::new();
         buffer.push_str(text);
+
+        // Shape the text using HarfBuzz.
         let buffer = font.owned_font_face.with_ref(|face| {
             makepad_rustybuzz::shape(face, &[], buffer)
         });
 
         let infos = buffer.glyph_infos();
+
+        // Modify the way glyph iteration is handled.
```

Comment:
This is not a very useful comment. It makes sense in the context of this review, but people who will look a this code later will wonder *how* the glyph iteration was modified and why.

Reply:
This comment should be fine, right?
```
// Track the processed text position to avoid reprocessing characters
// that have already been handled through font fallback
let mut skip_to = 0;
```

Reply:
> This comment should be fine, right?这个评论应该没问题吧？
> 
> ```
> // Track the processed text position to avoid reprocessing characters
> // that have already been handled through font fallback
> let mut skip_to = 0;
> ```

 I replaced it with this comment, and it has been committed.